api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1780,16 +1813,18 @@ type AzurePlatformSpec struct {
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 
-	// SecurityGroupName is the name of an existing security group on SubnetName. This field is provided as part of the
+	// SecurityGroupID is the ID of an existing security group on the SubnetID. This field is provided as part of the
 	// configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM).
 	//
-	// Example: if your Network Security Group ID is /subscriptions/<subscriptionID>/resourcegroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<securityGroupName>,
-	//          your SecurityGroupName is <securityGroupName>
+	// This security group is expected to exist under the same subscription as SubscriptionID. This resource is expected
+	// to exist in VnetResourceGroupName. If VnetResourceGroupName is not provided, HyperShift will assume it is in
+	// ResourceGroupName.
```

Comment:
I am not sure that this part of the description is correct:
```
If VnetResourceGroupName is not provided, HyperShift will assume it is in
ResourceGroupName.
```

`SecurityGroupID` is a full resource id. This means as part of its value it contains a resource group name part. Hypershift shouldn't assume where it is. Did you mean instead that "if VnetResourceGroupName is not provided, this resource is expected to exist in ResourceGroupName" ?

Reply:
This is OBE with the removal of VnetResourceGroupName